### PR TITLE
[test-framework] Add control RPC to test service

### PR DESCRIPTION
**Pull request by nmittler**
The RPC allows us to simulate in-mesh traffic from one service to
another.

Creating a copy of the pilot test service under pkg/test. This is a
more appropriate home for the code and also avoids changing the rest
of the code that depends on the old service.

### CODE REVIEW

File: pkg/test/service/echo/application.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/service/echo/proto"
+)
+
+// Application is a simple application than processes echo requetss via various transports.
```
Comment (ozevren) [inline pkg/test/service/echo/application.go]: requetss => requests

File: pkg/test/service/echo/application.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	TLSCert string
+	// TLSKey defines the server-side TLS key to use with GRPC.
+	TLSCKey string
+	// Version string
+	Version string
+
+	servers []serverInterface
+}
+
+// Start starts this application.
```
Comment (ozevren) [inline pkg/test/service/echo/application.go]: Minor nit: Can shorten to "Start the application.'"

**Comment by ozevren** [comment]
Presumably, this is all copied code and does not contain your additions, right?
/lgtm

File: pkg/test/service/echo/application.go (hunk 2 above)
Comment (nmittler) [reply pkg/test/service/echo/application.go]: done.

File: pkg/test/service/echo/application.go (hunk 1 above)
Comment (nmittler) [reply pkg/test/service/echo/application.go]: done.

**Comment by nmittler** [comment]
> Presumably, this is all copied code and does not contain your additions, right?

@ozevren yes, that's correct.  The only addition is the new handler method for the ForwardedEchoRequest

**Pull request merged**